### DISCUSSION TITLE
[Feature] Support for Docker Swarm

### DISCUSSION THREAD

**Discussion by bootsareme** [immich-app/immich#11459] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Immich as a service should be scalable to Docker Swarm. I want to distribute Immich over serveral nodes and make it fault tolerant and highly available. I have modified the Docker Compose file to work for Swarm. However, this makes Immich highly unstable and the database crashes frequently. Immich works great in regular Docker but I do not want it to be a single point of failure.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by mmomjian** [comment]
You should use an external postgres database for this that supports logical replication/fault tolerance. Postgres in Docker is not suitable for instant failover. This is not really within the scope of Immich.

**Reply by bo0tzz** [reply]
^ this. All the rest of Immich is already suitable for distribution/HA.
